src/offsetPath.js

```diff
@@ -18,12 +18,33 @@
       return undefined;
     } else if (ray !== null) {
       inputType = 'ray';
-      var rayInput = ray[1];
-      var rayInputDegrees = parseAngleAsDegrees(rayInput);
-      if (rayInputDegrees === null) {
-        return undefined;
+      var rayInput = ray[1].split(/\s+/);
+      var result = {type: inputType,
+                    input: 'none',
+                    contain: false,
+                    size: 'none'
+                   };
+
+      for (var i = 0; i < rayInput.length; i++) {
+        if (rayInput[i] === 'contain') {
+          result.contain = true;
+        } else if (rayInput[i] === 'closest-side') {
+          result.size = 'closest-side';
+        } else if (rayInput[i] === 'farthest-side') {
+          result.size = 'farthest-side';
+        } else if (rayInput[i] === 'closest-corner') {
+          result.size = 'closest-corner';
+        } else if (rayInput[i] === 'farthest-corner') {
+          result.size = 'farthest-corner';
```

Comment:
This can be much nicer and avoid duplicating magic strings and the assignment logic.